### PR TITLE
perf: Use weakmap for dom data

### PR DESCRIPTION
**Pull request by brandonocasey**
## Description
Use a weakmap so that we don't have to do the listener removal dance anymore on dispose.

### CODE REVIEW

**Comment by gkatsev** [comment]
Huh, apparently, WeakMap is supported in IE11. Though, it's only basic support, so, we'd want to double check that the operations we are using are supported in IE11.

File: src/js/component.js [hunk 1]
```diff
@@ -152,7 +151,6 @@ class Component {
         this.el_.parentNode.removeChild(this.el_);
       }
 
-      DomData.removeData(this.el_);
```
Comment (brandonocasey) [inline src/js/component.js]: We don't have to do this, once `this.el_` becomes null the weakmap will remove it on its own.

Comment (gkatsev) [reply src/js/component.js]: I don't think we should be relying on the behavior of the garbage collector, when we're done using it we should empty it out.

File: src/js/utils/events.js [hunk 2]
```diff
@@ -250,7 +253,11 @@ export function on(elem, type, fn) {
     return _handleMultipleEvents(on, elem, type, fn);
   }
 
-  const data = DomData.getData(elem);
+  if (!DomData.has(elem)) {
```
Comment (gkatsev) [inline src/js/utils/events.js]: I see a couple usages of if !has, set, then get. Do you think it's worth a helper?

Comment (brandonocasey) [reply src/js/utils/events.js]: I think that it might now make sense in this case since it is only used in two places.

**Comment by gkatsev** [comment]
This is good to go but I think I'm going to hold off on this to be a minor change.

File: src/js/utils/events.js (hunk 2 above)
Comment (gkatsev) [reply src/js/utils/events.js]: Yeah, you're right. If we end up with this a few more times, then it'll make sense.

**Comment by wind-stone** [comment]
Rollup did't compiler the WeapMap, as a result, there is an error "WeapMap is undefined" in Android 4.x

**Comment by gkatsev** [comment]
I have a PR to fix this pending here https://github.com/videojs/video.js/pull/6289

**Comment by gkatsev** [comment]
We merged #6289 and will be releasing it soon.

**Pull request merged**